### ISSUE TITLE
[Platformer] Changing Animations to one of a bigger canvas size moves the character based on the bounding box

### ISSUE THREAD

**Issue by Arnwaz** [4ian/GDevelop#4775]
When using the platformer character behavior when the animation is changed to an animation of a bigger canvas size, even if the actual sprite, collision box, and points are setup to be identical, then the character will be moved based on the bounding box regardless of points and the collision box. This only happens when the character is on the floor and they're not moved by a difference in the x size of the bounding box, if they're against a wall, only the y position is moved. 

  1. Create a sprite with the Platformer Character behavior and give it two identical animations, resize only the canvas of the second animation while leaving the size of the sprite the same. Be sure to set the collision box and points to the same spot on the sprites for an accurate test.
  2. Place the character in the scene on the floor and create an event which swaps the animation of the character to the one with a bigger canvas size.
  3. Watch the character "bounce" upwards despite the sprite size, collision box, and points being the same.
  4. Add an event to turn off the Platformer Character behavior, notice how the "bounce" no longer occurs. 

Minimalist version of the bug below:

[PlatformerBug.zip](https://github.com/4ian/GDevelop/files/10333413/PlatformerBug.zip)

Using GDevelop 5.1.155 on the Desktop app

Edit: I realized the correct term is bounding box.

**Comment by Bouh** [comment]
@D8H You had made modifications on the platformer several months (years?) ago, it may not be fresh anymore, but it may remind you of something?

**Comment by D8H** [comment]
This is because of this code:

https://github.com/4ian/GDevelop/blob/9e19aa888631c348106a15cfd81e842a3ceaced5/Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts#L1670-L1677

It uses the object height which is the height of the whole image. It should use the bonding box of the collision mask like the rest of the code.

**Comment by D8H** [comment]
As a workaround, using animation frames of the same size should work.

**Comment by 4ian** [comment]
@D8H do you think we could replace this bad repositioning code by one using the AABB? With a flag to keep compatibility with old games?

**Comment by D8H** [comment]
> @D8H do you think we could replace this bad repositioning code by one using the AABB?

Yes, just to clarify, it's the AABB of the collision mask not `getAABB()`. We should find a way to use the one evaluated for floor following.

> With a flag to keep compatibility with old games?

I think a compatibility is not needed as the existing code only work well for objects with a default collision mask and the fix will handle this case with the exact same behavior.
